var/spack/repos/builtin/packages/gnutls/package.py

```diff
@@ -39,72 +43,76 @@ class Gnutls(AutotoolsPackage):
     version("3.5.9", sha256="82b10f0c4ef18f4e64ad8cef5dbaf14be732f5095a41cf366b4ecb4050382951")
     version("3.3.9", sha256="39166de5293a9d30ef1cd0a4d97f01fdeed7d7dbf8db95392e309256edcb13c1")
 
-    depends_on("c", type="build")
-    depends_on("cxx", type="build")
-
+    # Variants
     variant("zlib", default=True, description="Enable zlib compression support")
     variant("zstd", default=True, description="Enable zstd compression support", when="@3.7:")
-    # See https://lists.gnupg.org/pipermail/gnutls-help/2023-February/004816.html
     variant("guile", default=False, description="Enable Guile bindings", when="@:3.7")
     variant(
         "brotli", default=True, description="Enable brotli compression support", when="@3.7.4:"
     )
 
-    # gnutls+guile is currently broken on MacOS.  See Issue #11668
-    conflicts("+guile", when="platform=darwin")
+    # Conflicts
+    conflicts(
+        "+guile",
+        when="platform=darwin",
+        msg="gnutls+guile is currently broken on MacOS. See Issue #11668",
+    )
+    conflicts("%clang@16:", when="@:3.7", msg="-Wimplicit-int is an error in newer clang versions")
+    conflicts(
+        "%apple-clang@15:",
+        when="@:3.7",
+        msg="-Wimplicit-int is an error in newer Apple clang versions",
+    )
```

Comment:
Does this work?
```suggestion
    conflicts(
        "%apple-clang@15:,clang@16:",
        when="@:3.7",
        msg="-Wimplicit-int is an error in newer Apple clang versions",
    )
```

Reply:
Ooo maybe, I'll try it out (probably tomorrow)

Reply:
```
==> Error: cannot load package 'gnutls' from the 'builtin' repository: unexpected characters in the spec string
%apple-clang@15:,%clang@16:
```

This does not work.

Reply:
Unfortunately this doesn't work. It seems to ignore the conflicts statement entirely when given two compilers.

(Could be a bug I'll pass on.)